cli/main.rs

```diff
@@ -397,6 +397,21 @@ fn run_script(flags: DenoFlags, argv: Vec<String>) {
   }
 }
 
+fn version_command() {
+  // The TypeScript version is only available in JS at the moment.
+  // TODO: Rewrite if this changes.
+  eval_command(
+    DenoFlags::default(),
+    vec![
+      "eval".to_string(),
+      "console.log(\"deno:\", Deno.version.deno);
+       console.log(\"v8:\", Deno.version.v8);
+       console.log(\"typescript:\", Deno.version.typescript);"
+        .to_string(),
+    ],
+  );
+}
```

Comment:
I don't know. I'm not sure this is much better. Ideally we would have access to the typescript version in Rust, and we could just print without launching an isolate.

Reply:
@ry Yeah. I guess the purpose here is to keep subcommand branching/handling out of JS and leave it as a self contained hack in `main.rs`. Especially if the ideal fix isn't going to be available any time soon.

Though I see that xeval and the repl are still half handled in `denoMain()` anyway :(. I'd try and have those modules be injected by `main.rs` when needed as well, but I'm not sure. `main.ts` is confusing as it is and I don't like that subcommands are handled partially at both ends.

Reply:
> Ideally we would have access to the typescript version in Rust, and we could just print without launching an isolate.

Make it another const in Rust? That's one more thing to change manually during TypeScript upgrade but it might be worth that instead of spinning up the isolate just to print of version.

> Though I see that xeval and the repl are still half handled in `denoMain()` anyway :(. I'd try and have those modules be injected by `main.rs` when needed as well, but I'm not sure. `main.ts` is confusing as it is and I don't like that subcommands are handled partially at both ends.

Once new snapshotting infrastructure lands `xeval` and `repl` can be rewritten to have their own entry point scripts instead of special handling in `main.rs` and `main.ts`. That would be pretty elegant solution that would allow to move `xeval` and `repl` into `deno_std` instead of handling them specially in `cli`.

I don't know, just thinking out loud.

Reply:
We could maybe regex it out of `node_modules/typescript/lib/typescript.js`:
```js
   // WARNING: The script `configureNightly.ts` uses a regexp to parse out these values.
    // If changing the text in this section, be sure to test `configureNightly` too.
    ts.versionMajorMinor = "3.5";
    /** The version of the TypeScript compiler release */
    ts.version = ts.versionMajorMinor + ".1";
```
I don't know if that complexity is worth it though. In https://github.com/denoland/deno/pull/2825 ts.version is handled with some hacks, having it the value in Rust would clean up more. Let's wait for that to land and then maybe add a regex.

Reply:
> move `xeval` and `repl` into `deno_std`

Now that I think about it, isn't that already possible with `xeval`? The only special handling it gets is `worker.execute()` - is there a reason we couldn't use JS `eval()` or something?

Reply:
Yeah, we could definitely startup an isolate to extract it. That's probably better. It could probably be done in `cli_snapshots/build.rs` (that's a file introduced in #2825)